packages/tools/src/tools/ScaleOverlayTool.ts

```diff
@@ -0,0 +1,666 @@
+    annotation: ScaleOverlayAnnotation;
+  } | null = {} as any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      configuration: {
+        viewportId: '',
+        scaleLocation: 'bottom',
+        scaleColor: 'yellow',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  _init = (): void => {
+    const renderingEngines = getRenderingEngines();
+    const renderingEngine = renderingEngines[0];
+
+    if (!renderingEngine) {
+      return;
+    }
+
+    let viewports = renderingEngine.getViewports();
+    viewports = filterViewportsWithToolEnabled(viewports, this.getToolName());
+
+    let viewport = viewports[0];
+
+    if (this.configuration.viewportId) {
+      viewport = viewports.find(
+        (viewportId) => viewportId.id === this.configuration.viewportId
+      );
+    }
+
+    if (!viewport) {
+      return;
+    }
+
+    const { element } = viewport;
+    const { viewUp, viewPlaneNormal } = viewport.getCamera();
+
+    const viewportCanvasCornersInWorld =
+      csUtils.getViewportImageCornersInWorld(viewport);
+
+    let annotation = this.editData.annotation;
+
+    if (!annotation) {
+      const newAnnotation: ScaleOverlayAnnotation = {
+        metadata: {
+          toolName: this.getToolName(),
+          viewPlaneNormal: <Types.Point3>[...viewPlaneNormal],
+          viewUp: <Types.Point3>[...viewUp],
+          FrameOfReferenceUID: viewport.getFrameOfReferenceUID(),
+          referencedImageId: null,
+        },
+        data: {
+          handles: {
+            points: viewportCanvasCornersInWorld,
+          },
+        },
+      };
+
+      addAnnotation(element, newAnnotation);
+      annotation = newAnnotation;
+    } else {
+      this.editData.annotation.data.handles.points =
+        viewportCanvasCornersInWorld;
+    }
+
+    this.editData = {
+      viewport,
+      renderingEngine,
+      annotation,
+    };
+  };
+
+  onSetToolEnabled = (): void => {
+    this._init();
+  };
+
+  onCameraModified = (evt: Types.EventTypes.CameraModifiedEvent): void => {
+    // If the camera is modified, we need to update the viewport
+    // that the camera was modified on
+    this.configuration.viewportId = evt.detail.viewportId;
+    this._init();
+  };
+
+  /**
+   * Used to draw the scale annotation in each request animation
+   * frame.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   * @returns
+   */
+
+  renderAnnotation(
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ) {
+    if (!this.editData.viewport) {
+      return;
+    }
+    const location = this.configuration.scaleLocation;
+    const { annotation, viewport } = this.editData;
+    const image = viewport.getImageData();
+    const imageId = viewport.getCurrentImageId();
```

Comment:
if we are using imageId, that means only stackViewport is supported. Can you at least mention it in the documentation? and in the demo doc?

Reply:
Removed this section from the code, the tool works on both stack and volume viewports as shown below.

![scaleoverlay-volume](https://user-images.githubusercontent.com/9090600/216643568-1336f4a0-a60d-4f13-9be0-d05f5a0df07b.gif)